src/pages/index.js

```diff
@@ -0,0 +1,264 @@
+import "../pages/index.css";
+import { Card } from "../components/Card.js";
+import { FormValidator } from "../components/FormValidator.js";
+import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithImage from "../components/PopupWithImage.js";
+import UserInfo from "../components/UserInfo.js";
+import Section from "../components/Section.js";
+import {
+  initialCards,
+  editButton,
+  addButton,
+  profileForm,
+  nameInput,
+  jobInput,
+  cardForm,
+  updateAvatar,
+  avatarForm,
+  avatarImage,
+  profileName,
+  profileJob,
+  config,
+} from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
+
+const editFormValidator = new FormValidator(config, profileForm); //creating a new var for the edit modal using the FormValidator class
+const addCardFormValidator = new FormValidator(config, cardForm);
+const avatarUpdateValidator = new FormValidator(config, avatarForm);
+
+const newEditPopup = new PopupWithForm(
+  ".profile-modal",
+  handleProfileFormSubmit
+); //creating a new var for edit form from the popupWithForm class
+newEditPopup.setEventListeners();
+
+const newCardPopup = new PopupWithForm(".card-modal", addCardElement);
+//creating a new var for the card form from the popupWithFrom class
+newCardPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm(".avatar-modal", upDateAvatar);
+avatarPopup.setEventListeners();
+
+const newImagePopup = new PopupWithImage(".image-modal");
+newImagePopup.setEventListeners();
+
+const sectionCards = new Section(
+  { items: initialCards, renderer: createCard },
+  ".cards"
+);
+
+const userInfo = new UserInfo(
+  ".profile__name",
+  ".profile__descripton",
+  ".profile__img"
+);
+
+const confirmationPopup = new PopupWithConfirmation(".confirmation-modal");
+confirmationPopup.setEventListeners();
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "bfb869e8-08ec-4b67-8cc1-518f5a35ed9e",
+    "Content-Type": "application/json",
+  },
+});
+
+////////////////////////////////////////////////////// functions ///////////////////////////////////////////////////////////
+
+//function that opening the img popup
+function handleImageClick(card) {
+  newImagePopup.open(card);
+}
+
+function createCard(cardData) {
+  return new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButtonClick,
+    handleLikedButtonClick
+  ).getView();
+}
+
+function addCardElement(cardData) {
+  newCardPopup.setLoading(true);
+  api
+    .addNewCard(cardData.title, cardData.link, cardData._id)
+    .then((cardData) => {
+      sectionCards.addItem(createCard(cardData));
+      addCardFormValidator.resetValidation();
+      newCardPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error in addNewCard:", error);
+      return Promise.reject(error);
```

Comment:
`Promise.reject` is not needed in block `catch` because it **throws** the error again outside the chain. The final `catch` should keep the error inside

**Please, correct it everywhere**

Reply:
fixed it i see what you mean 
for example if i get error 400 i will get it twice